### DISCUSSION TITLE
FEAT: Add ability to style hover badges independently of selected badges

### DISCUSSION THREAD

**Discussion by TzviPM** [GrapesJS/grapesjs#4152] [Ideas]
Currently, when a component is hovered, there is a badge displayed in a toolbar. I’d like to persist that badge when the component is selected and make it appear differently from the hovered badge.

Possible solutions:

1. provide an extension point to the SelectComponent command to more easily control this. For example, encapsulate the `display: none` logic when changing focus into a helper method that can be overridden with `commands.extend`
2. decouple the display badge from the remaining  tools on a component and provide a css class to style them differently.

Please let me know if I missed anything and there’s a reasonable way to do this currently.

**Comment by artf** [comment]
Yeah, I'm planning to refactor the canvas in order to allow higher customizability of component rects but refactoring the SelectComponent for better extendability probably might be the fastest temporary solution. So, if anyone is willing to help, PRs are welcome.
